legacy_promotions/lib/solidus_legacy_promotions/engine.rb

```diff
@@ -1,11 +1,19 @@
 # frozen_string_literal: true
 
 require 'solidus_legacy_promotions'
-
+require 'flickwerk'
 module SolidusLegacyPromotions
   class Engine < ::Rails::Engine
     include SolidusSupport::EngineExtensions
 
+    include Flickwerk
+
+    initializer "solidus_legacy_promotions.patch_state_machine" do
+      config.to_prepare do
+        SolidusLegacyPromotions::SpreeOrderStateMachinePatch
```

Comment:
This is necessary because `Spree::StateMachines::Order` is defined in `lib` and not autoloaded, so that Flickwerk's `on_load` hook won't do the trick.